kratos/utilities/nurbs_utilities/snake_sbm_utilities.cpp

```diff
@@ -0,0 +1,872 @@
+        ModelPart& rSkinModelPart, 
+        int rEchoLevel, 
+        Vector& knotVectorU, 
+        Vector& knotVectorV,
+        const std::size_t numberOfInnerLoops,
+        const double lambdaInner,
+        const double lambdaOuter)
+    {   
+
+        // Initilize the property of skin_model_part_in and out
+        if (rSkinModelPartInnerInitial.NumberOfNodes()>0) {
+            // template argument IsInnerLoop set true
+            CreateTheSnakeCoordinates<true>(rIgaModelPart, rSkinModelPartInnerInitial, rSkinModelPart, knotVectorU, knotVectorV, numberOfInnerLoops, lambdaInner);
+                
+        }
+        if (rSkinModelPartOuterInitial.NumberOfNodes()>0) {
+            // template argument IsInnerLoop set false
+            CreateTheSnakeCoordinates<false>(rIgaModelPart, rSkinModelPartOuterInitial, rSkinModelPart, knotVectorU, knotVectorV, 1, lambdaOuter);
+        }
+    }   
+
+    template <bool TIsInnerLoop>
+    void SnakeSbmUtilities::CreateTheSnakeCoordinates(
+        ModelPart& rIgaModelPart, 
+        ModelPart& rSkinModelPartInitial,
+        ModelPart& rSkinModelPart, 
+        Vector& knotVectorU, 
+        Vector& knotVectorV,
+        const std::size_t numberOfLoops,
+        const double lambda) 
+    { 
+        int r_echo_level = GetEchoLevel();
+        const bool is_inner = TIsInnerLoop;
+    
+        std::string surrogate_sub_model_part_name; 
+        std::string skin_sub_model_part_name; 
+        // ModelPart skin_sub_model_part; 
+        if (is_inner)  {
+            surrogate_sub_model_part_name = "surrogate_inner";
+            skin_sub_model_part_name = "inner";
+        }
+        else {
+            surrogate_sub_model_part_name = "surrogate_outer";
+            skin_sub_model_part_name = "outer";
+        }
+        
+        ModelPart& r_skin_sub_model_part = rSkinModelPart.GetSubModelPart(skin_sub_model_part_name);
+        ModelPart& r_surrogate_sub_model_part = rIgaModelPart.GetSubModelPart(surrogate_sub_model_part_name);
+
+        if (!rSkinModelPartInitial.HasProperties(0)) rSkinModelPartInitial.CreateNewProperties(0);
+        if (!rSkinModelPart.HasProperties(0)) rSkinModelPart.CreateNewProperties(0);
+        auto p_cond_prop_in = rSkinModelPartInitial.pGetProperties(0);
+        rSkinModelPartInitial.AddProperties(p_cond_prop_in);
+
+        array_1d<double, 2> knot_step_uv(2);
+        knot_step_uv[0] = std::abs(knotVectorU[std::ceil(knotVectorU.size()/2) +1]  - knotVectorU[std::ceil(knotVectorU.size()/2)] ) ;
+        knot_step_uv[1] = std::abs(knotVectorV[std::ceil(knotVectorV.size()/2) +1]  - knotVectorV[std::ceil(knotVectorV.size()/2)] ) ;
+
+        Vector mesh_sizes_uv(2);
+        mesh_sizes_uv[0] = knot_step_uv[0]; 
+        mesh_sizes_uv[1] = knot_step_uv[1];
+        auto& surrogate_model_part = rIgaModelPart.GetSubModelPart(surrogate_sub_model_part_name);
+        surrogate_model_part.GetProcessInfo().SetValue(MARKER_MESHES, mesh_sizes_uv);
```

Comment:
I wouldn't save this in the `ProcessInfo`, specially with a meaningless variable name such as `MARKER_MESHES`.

Instead, I'd move the implementation of the `mesh_sizes_uv` calculation to an `IGAUtilities` or `KnotSpanUtilities` and call it where it needed as this is an operation with negligible cost. In here I assume that we may want to reuse this in the future.

Note that current approach is far from a transparent usage.